sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ActionDescriptorFactory.scala

```diff
@@ -19,106 +19,131 @@ package kalix.springsdk.impl
         KalixMethod(serviceMethod).withKalixOptions(buildJWTOptions(serviceMethod.javaMethod))
       }
 
     //TODO make sure no subscription should be exposed via REST.
+    // methods annotated with @Subscribe.ValueEntity
     import ReflectionUtils.methodOrdering
-    val subscriptionValueEntityMethods = component.getMethods
+    val subscriptionValueEntityMethods: IndexedSeq[KalixMethod] = component.getMethods
       .filter(hasValueEntitySubscription)
       .sorted // make sure we get the methods in deterministic order
       .map { method =>
         val subscriptionOptions = eventingInForValueEntity(method)
         val kalixOptions =
           kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
 
-        KalixMethod(SubscriptionServiceMethod(method))
+        KalixMethod(PubSubServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
+      .toIndexedSeq
 
-    val subscriptionEventSourcedEntityMethods = component.getMethods
+    // methods annotated with @Subscribe.EventSourcedEntity
+    val subscriptionEventSourcedEntityMethods: IndexedSeq[KalixMethod] = component.getMethods
       .filter(hasEventSourcedEntitySubscription)
       .sorted // make sure we get the methods in deterministic order
       .map { method =>
-        if (hasEventSourcedEntitySubscription(component))
-          throw InvalidComponentException(
-            s"You cannot use Subscribe annotation in both the methods and the class. You can do either one or the other.")
         val subscriptionOptions = eventingInForEventSourcedEntity(method)
         val kalixOptions =
           kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
-
-        KalixMethod(SubscriptionServiceMethod(method))
+        KalixMethod(PubSubServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
+      .toIndexedSeq
 
-    val subscriptionEventSourcedEntityClass =
+    val subscriptionEventSourcedEntityClass: Map[String, Seq[KalixMethod]] =
       if (hasEventSourcedEntitySubscription(component)) {
-        component.getMethods.sorted // make sure we get the methods in deterministic order
-          .filter(hasActionOutput)
-          .collect { case method =>
-            val subscriptionOptions = eventingInForEventSourcedEntity(component)
-            val kalixOptions =
-              kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+        val kalixMethods =
+          component.getMethods
+            .filter(hasActionOutput)
+            .sorted // make sure we get the methods in deterministic order
+            .map { method => KalixMethod(PubSubServiceMethod(method)) }
+            .toSeq
 
-            KalixMethod(SubscriptionServiceMethod(method))
-              .withKalixOptions(kalixOptions)
-          }
-      } else Array.empty[KalixMethod]
+        val entityType = findEventSourcedEntityType(component)
+        Map(entityType -> kalixMethods)
```

Comment:
There some structural changes here to make it possible to have service level annotations for subscription. That's needed when publishing to stream